Fixes second callback after timeout when leading: true

idpaterson opened the pull request:
I noticed while working on the [context pull request](https://github.com/bjoerge/debounce-promise/pull/3) test cases that my debounced function was actually getting called twice when `leading: true`. This was not caught in the current tests because the promise resolves immediately  before the debounce timeout has fully completed.

This pull request uses `clear` for the timeout when `leading: true` rather than `run`. The latter was calling the function a second time after the timeout elapsed.

bjoerge commented:
Hi @idpaterson, thanks for the pull request!

I'm pretty sure the correct/expected behavior is to invoke the function twice when `leading: true`. Once immediately and once after the wait period is over. This is also how lodash.debounce works:

```js
const {debounce} = require('lodash')
const run = debounce(function run() {
  console.log('called')
}, 200, {leading: true})

run()
run()
setTimeout(run, 100)
```
Will output:
```
called
called
```

In cases where this is _not_ the desired behavior it could instead be covered by a `trailing` option that, if set to `false`, only invokes the given function once, at the leading end.

idpaterson commented:
Ah you're absolutely correct, the test that I added did not actually test the problem that I was seeing. The original issue I noticed was that the callback is called again *even if the debounced function is called only once*. This was certainly a bad fix on my part. 

I updated the pull request with two tests, one the exact opposite of my first ensuring that the callback *is* called again if the debounced function is invoked multiple times during the timeout, and one to ensure that if invoked only once the callback is called only once.

bjoerge commented:
Ah, that makes a lot more sense. Thanks a lot for this!

bjoerge commented:
Published as v2.0.0 as this could potentially be a breaking change

idpaterson commented:
Thank you for the speedy publish!
